app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -20,18 +19,21 @@ module AuthnAzure
       def_delegators :@authenticator_input, :service_id, :authenticator_name, :account, :username, :request, :credentials
 
       def call
-        true
+        validate_application_identity
       end
 
       private
 
-      #def validate_azure_token
-      #
-      #end
-      #
-      #def validate_application_identity
-      #
-      #end
+      # expecting to receive xms_mirid and oid as inputs (either global instance or function). This will change to hash map.
+      def validate_application_identity
+        @validate_application_identity.(
+            service_id: service_id,
+            account: account,
+            username: username,
+            xms_mirid: decoded_credentials["xms_mirid"],
+            oid: decoded_credentials["oid"]
```

Comment:
I added methods to get these variables so we also log the output to the log. Please change to:
```
xms_mirid: xms_mirid,
oid: oid
```